### PR TITLE
Validate IPFS Metadata against a JSON schema

### LINKED ISSUE #98: Validate IPFS metadata
We can't trust the meta data coming from IPFS.

We should validate the data against a JSON schema and ignore jobs that fail validation.

### PR DESCRIPTION
**Pull request by deweller**
Also refactors logic of PostJobModal into a usePostJob hook

Closes #98 

This is a stacked PR

### CODE REVIEW

File: frontend/src/components/create-contract/ICreateContractForm.ts [hunk 1]
```diff
@@ -8,6 +8,6 @@ export interface ICreateContractForm {
   labels: IListBoxItem[];
   identity: IListBoxItem[];
   acceptanceTests: IListBoxItem[];
-  deposit: string;
+  requiredDeposit: string;
```
Comment (reisr3) [inline frontend/src/components/create-contract/ICreateContractForm.ts:11]: Why are we making the distinction between `requiredDeposit` and `deposit` now?

File: frontend/src/components/smart-contracts/modals/hooks/usePostJob.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        metadata,
+      });
+
+      setIPFSCid(result.ipfsCid);
+      setShowIPFSModal(false);
+    };
+
+    // clear error message when closing the modal
+    if (!showPostJobModal) {
+      setErroMessage('');
```
Comment (reisr3) [inline frontend/src/components/smart-contracts/modals/hooks/usePostJob.ts:89]: Typo

File: frontend/src/pages/Contracts.tsx [hunk 3]
```diff
@@ -8,7 +8,15 @@ const Contracts = () => {
...
     <>
       <SortFilterHeading heading="Contracts" displayCreate />
-      {isLoading ? <div>Loading..</div> : <ContractsContainer jobs={jobs} />}
+      {isLoading ? (
+        <div>Loading..</div>
+      ) : jobs.length ? (
+        <ContractsContainer jobs={jobs} />
+      ) : (
+        <div>
+          <div>No jobs are available to show at this time.</div>
```
Comment (reisr3) [inline frontend/src/pages/Contracts.tsx:17]: Probably don't need this extra wrapping div, right?

Also, maybe this message is better as `No jobs found.`

File: frontend/src/utils/schema.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  const schema = loadSchemaByVersionNumber(schemaVersion);
+  if (!schema) {
+    return {
+      isValid: false,
+      error: 'Schema version not found',
+    };
+  }
+
+  const validater = ajv.compile(schema);
+  const isValid = validater(metadata);
```
Comment (reisr3) [inline frontend/src/utils/schema.ts:29]: sp: Validator

File: frontend/src/components/single-contract/StartJobButton.tsx [hunk 5]
```diff
@@ -34,7 +35,10 @@ export const StartJobButton: React.FC = () => {
 
         <StartJobModal
           jobId={data.id}
-          deposit={data.deposit}
+          deposit={multiplyByIntegerPercentage(
```
Comment (reisr3) [inline frontend/src/components/single-contract/StartJobButton.tsx:38]: This is just for display purposes right? Like we're calculating the actual percentage in the contract?

**Review by reisr3 — CHANGES_REQUESTED** [review]
Looking great! Just a couple of things.

Also, generally I think we should look at cleaning up `useJob`. It's kinda hard to follow what is happening in there and its gotten a bit unwieldy. Maybe we pair on that sometime this week? I can explain my thoughts better.

File: frontend/src/components/single-contract/StartJobButton.tsx (hunk 5 above)
Comment (deweller) [reply frontend/src/components/single-contract/StartJobButton.tsx:38]: This code is replaced in the next PR.

File: frontend/src/components/smart-contracts/modals/hooks/usePostJob.ts (hunk 2 above)
Comment (deweller) [reply frontend/src/components/smart-contracts/modals/hooks/usePostJob.ts:89]: I'm not seeing it.  Am I missing something obvious here?

**Comment by deweller** [comment]
> Looking great! Just a couple of things.
> 
> Also, generally I think we should look at cleaning up `useJob`. It's kinda hard to follow what is happening in there and its gotten a bit unwieldy. Maybe we pair on that sometime this week? I can explain my thoughts better.

Let's do it.

File: frontend/src/components/create-contract/ICreateContractForm.ts (hunk 1 above)
Comment (deweller) [reply frontend/src/components/create-contract/ICreateContractForm.ts:11]: In the smart contract, there is a job attribute called deposit that is the actual amount sent by the engineer.  In an attempt to avoid confusion, I renamed this requiredDeposit which is the minimum deposit required.

In the next PR, requiredDeposit is no longer a metadata field and becomes an attribute in the smart contract data.
